### ISSUE TITLE
I cannot deploy WASM to IIS

### ISSUE THREAD

**Issue by TopperDEL** [unoplatform/uno#16015]
### Current behavior

I have multiple problems. Either the publish fails with "file is locked" or the published app is not accessable or - the current error - it states that "there is no target for net8.0/win-x64".

I'll start with the "no target"-error:
[msbuild.zip](https://github.com/unoplatform/uno/files/14740243/msbuild.zip)

This is a follow-up-issue from #14637 .

### Expected behavior

The deploying of a freshly created WASM-app to IIS should work.

### Workaround

None, yet

### Works on UWP/WinUI

Yes

### Affected platforms

WebAssembly

### IDE

Visual Studio 2022

**Comment by TopperDEL** [comment]
I just created a new app, tried Windows (works), tried WASM locally (works), created a publish profile and tried to publish and got the "IISDEployTest.Wasm.dll is used by another process".

The binlog for this error (it is a different project then the above!) is here: 
[msbuild.zip](https://github.com/unoplatform/uno/files/14740320/msbuild.zip)

**Comment by TopperDEL** [comment]
Now I managed to get a deployment working. The site in IIS now looks like this:

![image](https://github.com/unoplatform/uno/assets/1833242/6e6fa953-d9c6-4d00-a8e4-772fb2022969)

It feels like the site is using the wrong directory. But as mentioned in #14637 I see no other way to create a site and deploy to it.

**Comment by TopperDEL** [comment]
Ok, I just realized, that the template also creates a server project that serves the WASM-App, too. So I do not need to deploy the WASM-project itself but I can simply deploy the Server-project. That works flawlessly as expected!

So if this is the way to go, then I'm fine. If WASM still should be deployable by its own (what I assume, though), the above errors are still there.

But for me, the server-project is good enough and works!

**Comment by jeromelaban** [comment]
Thanks for the update @TopperDEL. The wasm project should be deployable indeed. I may have missed this, but did you try using `dotnet publish` from the command line?

**Comment by TopperDEL** [comment]
Yes, I did. But it lead to nearly the same issue - though I'm not quite sure if it was due to missing password or so. So maybe the reason for failing deploy was different.

**Comment by TopperDEL** [comment]
I just stumbled across this problem again. I could not publish the WASM-Head to IIS. Then I tried to do it using the terminal and `dotnet publish`. That lead me to an issue: I have one main uno-app (already on 5.2) that is using central package management. The app I would like to deploy now is in a subfolder of the other app and is on Uno 5.1 and is NOT using central package management. But if I do not define that explicitly, it found the `Directory.Packages.props`from the 5.2-app and I got the error that I must not define the package version when using the central package management.
I could overcome this by explicitly set `<ManagePackageVersionsCentrally>false</ManagePackageVersionsCentrally>`.

Then the publish with `dotnet publish` worked, but I still cannot access the app. "You are not allowed to access the directory or page". I think there is some web.config-setting missing now.
I am still unable to publish it using Visual Studio. There the publish fails with "project.assets.json does not contain a target for net8.0/win-x64".

**Comment by TopperDEL** [comment]
After adding `<RuntimeIdentifiers>win-x64</RuntimeIdentifiers>` to my WASM-head I could deploy it, but still have issues with the app not being accessable. I think I'm missing some mime-types or so in the ISS-site-config.

What really confuses me is [this doc](https://platform.uno/docs/articles/how-to-host-a-webassembly-app.html#iis). I do not know how to set the physical path. If I create a site I need to define which folder it uses. Then I deploy the app and would expect the folder to contain the app itself - not necessarily in a subfolder. So there is no other way to "set a physical path" to another subfolder of my site.

**Comment by TopperDEL** [comment]
Now I got it! the physical path of my site can be changed in the "extended settings". There I simply have to set it to the "wwwroot"-folder underneath my site-path.

Now it is working as expected! I can deploy and the site is accesable again!

I would suggest to update the [doc](https://platform.uno/docs/articles/how-to-host-a-webassembly-app.html#iis), though. The physical path has to be set to "wwwroot", not to any "dist"-folder.

**Comment by TopperDEL** [comment]
Sorry, I have to re-open this.

Every consecutive deploy now writes below the "wwwroot"-physical path in IIS. So changing the physical path for the site is _not_ the way to go.

I create a new site:
c:\inetpub\wwwroot\myapp\

I deploy my site using VS:
c:\inetpub\wwwroot\myapp\wwwroot

I change the physical path and re-deploy - making the app unusable again:
c:\inetpub\wwwroot\myapp\wwwroot\wwwroot

The problem is: I need a web.config in the myapp-folder that configures the redirection to the wwwroot. This should be created by the deployment-task - or the deployment should not create a "deeper" path (appending "wwwroot").

**Comment by jeromelaban** [comment]
Thanks for the updates. Could you create a step-by-step repro scenario that shows the issue? I'm also wondering if it's something that got fixed in 5.2 somehow with the single project.

**Comment by TopperDEL** [comment]
So @jeromelaban, I just made an end-to-end-test and collected all steps I've done.

First, let's create a new app:
dotnet new unoapp -o iistest
 
And create a new site in IIS:
![image](https://github.com/unoplatform/uno/assets/1833242/c8346086-00ff-407b-acd8-3ed2247319b2)

Then I started win-head => working
I started wasm (iis express) => not working (obscure error)
I started wasm => working
I started wasm (iis express) => working (maybe the first run needs some initialization? - whatever)
 
Let's create a publish profile:
![image](https://github.com/unoplatform/uno/assets/1833242/36ad0b20-490c-4a03-a0f7-3bdb1948364d)
![image](https://github.com/unoplatform/uno/assets/1833242/8f911a52-08a3-4e4f-b856-e2a1046f8878)
![image](https://github.com/unoplatform/uno/assets/1833242/cd48d990-3679-46d2-b3b9-ec6cd33aa15f)
 
Started the deployment=> it fails with “…\iistest\obj\Release\net8.0-browserwasm\iistest.dll” is already in use by another process
Cleanup solution, deploy again => same error
Cleanup solution, closing VS, open Solution again, deploy again without any building in between => successful deployment
 
![image](https://github.com/unoplatform/uno/assets/1833242/68216b86-c4e4-4d7b-85ed-ddae5623c8f1)

BUT: the website is not available:
 
![image](https://github.com/unoplatform/uno/assets/1833242/5446be16-70ab-4c8d-a2e6-2d7aba57b71a)

“You are not allowed to view dies folder or site”
The reason: there is no document to serve in the root-folder of the IIS-site:
 
![image](https://github.com/unoplatform/uno/assets/1833242/4dda634b-1669-4551-9f04-fab2a4be25e3)

Now if I change the physical path of the site from this:
 
![image](https://github.com/unoplatform/uno/assets/1833242/5790857b-e45b-408a-b1df-204fb72bcac1)

To this:
 
![image](https://github.com/unoplatform/uno/assets/1833242/154f398c-1027-4125-b1e8-9d815b0db2a6)

It works:
 
![image](https://github.com/unoplatform/uno/assets/1833242/26f5cfdf-6099-44a4-be17-7f97a2af5ea2)

Now let’s deploy again from VS (cleanup, closing, re-opening, deploying) – we then get another wwwroot underneath the “iistest\wwwroot”-folder and we don’t get the latest version served from IIS (as that one is one folder deeper).
The folder structure now looks like this:
 
![image](https://github.com/unoplatform/uno/assets/1833242/523a2e85-5f66-4266-8e5a-c346c5630054)

This is exactly the error I get all the time. It seems the deployment has a wrong "ground" to start. Why does it create another wwwroot underneath my site-path? It should just deploy all files at the root. I cannot overcome this by chaning the physical path of the site as every subsequent deploy would again create a new wwwroot underneath.

**Comment by RaceAlex69** [comment]
Same identical problem , any updates ?

**Comment by jeromelaban** [comment]
@RaceAlex69 are you using the command line tooling to deploy? Also, are you using net9?
